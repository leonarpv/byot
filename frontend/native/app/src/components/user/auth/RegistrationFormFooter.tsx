import * as React from 'react';
import {StyleSheet, View} from 'react-native';
import {useTranslation} from 'react-i18next';
import {Button} from '../../elements/lib/Button';
import {Colors, Text} from 'react-native-ui-lib';

interface Props {
  onLogInPress: () => void;
}

interface State {}

const makeStyles = (props: Props, state: State) =>
  StyleSheet.create({
    root: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      marginTop: 10,
    },
    haveAccount: {
      color: Colors.grey50,
      marginTop: 12,
    },
  });

export const RegistrationFormFooter: React.FC<Props> = (props: Props) => {
  const styles = makeStyles(props, {});
  const {t} = useTranslation();
  return (
    <View style={styles.root}>
      <Text style={styles.haveAccount}>{t('Already have an account?')}</Text>
      <Button onPress={props.onLogInPress} variant="primary" link label={t(`Log in`)} />
    </View>
  );
};
