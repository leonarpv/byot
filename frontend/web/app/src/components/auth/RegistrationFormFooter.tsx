import * as React from 'react';
import {Box, makeStyles, Theme, Typography, WithStyles} from '@material-ui/core';
import {Button} from '@byot-frontend/web-common/src/components/elementary/form/Button';
import {Link} from 'react-router-dom';
import {Router} from '../../router/Router';
import {useTranslation} from 'react-i18next';

interface Props extends Partial<WithStyles<typeof styles>> {}

const styles = (theme: Theme) => ({
  alreadyHaveLabel: {
    color: theme.palette.grey[500],
  },
});
const useStyles = makeStyles(styles);

export const RegistrationFormFooter: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  const {t} = useTranslation();
  return (
    <Box display="flex" flexDirection="row" justifyContent="space-between" mt={4}>
      <Typography classes={{root: styles.alreadyHaveLabel}}>{t('Already have an account?')}</Typography>
      <Button color="secondary" variant="text" component={Link} to={Router.login.URI()}>
        {t('Log in')}
      </Button>
    </Box>
  );
};
