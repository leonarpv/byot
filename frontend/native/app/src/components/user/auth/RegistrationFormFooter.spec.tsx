import * as React from 'react';
import {render, wait, fireEvent} from '@testing-library/react-native';
import {RegistrationFormFooter} from './RegistrationFormFooter';

describe('<RegistrationFormFooter/>', () => {
  it('should render', async () => {
    const onLogInPress = jest.fn();
    const {asJSON, getByTestId} = render(<RegistrationFormFooter onLogInPress={onLogInPress} />);
    fireEvent.press(getByTestId('auth-registrationFormFooter-login'));
    await wait(() => expect(onLogInPress).toBeCalled());
    expect(asJSON()).toMatchSnapshot();
  });
});
