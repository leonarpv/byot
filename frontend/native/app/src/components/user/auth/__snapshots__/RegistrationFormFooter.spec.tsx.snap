// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationFormFooter/> should render 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FAFAF1",
          "marginTop": 12,
        }
      }
    >
      Already have an account?
    </Text>
    <TouchableOpacity
      accessibilityLabel="auth-registrationFormFooter-login"
      label="Log in"
      link={true}
      testID="auth-registrationFormFooter-login"
      variant="primary"
    />
  </View>
</View>
`;
